src/components/utils/test/thread_test.cc

```diff
@@ -72,6 +74,76 @@ class TestDelegate : public threads::ThreadDelegate {
   bool check_value_;
 };
 
+class InfiniteMainTestDelegate : public threads::ThreadDelegate {
+ public:
+  InfiniteMainTestDelegate()
+      : notified_(false), notify_running_(false), check_value_(false) {}
+
+  void threadMain() {
+    {
+      AutoLock test_lock(test_mutex_);
+      check_value_ = true;
+      notify_running_ = true;
+      cond_var_.Broadcast();
+    }
+
+    while (!notified_) {
+      usleep(10);
```

Comment:
@pestOO No. I haven't run them all but they should work everywhere. I am still trying to find the usernames of my new reviewers to add them.